src/assets/TokenDetectionController.ts

```diff
@@ -96,25 +106,57 @@ export class TokenDetectionController extends BaseController<
       networkType: MAINNET,
       selectedAddress: '',
       tokens: [],
+      disabled: true,
+      chainId: NetworksChainId.mainnet,
```

Comment:
Pre-existing problem, but, initializing this to be set to mainnet might leave this misconfigured on startup until the first network state change. We don't emit state changes on startup, so this could be bad.

We could address this by adding a `getNetworkState` method to the constructor, and asking what the initial state is.

Maybe better done in a separate PR though, since this one is already rather complex.

Reply:
Updated this PR to handle this. Using both `getNetworkState` and `getPreferencesState` to set defaults in config